File: .cpp/238.ProductofArrayExceptSelf.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int>left_product (n,1);
        vector<int>right_product (n,1);
        for (int i =1;i<n;i++)
        {
            left_product[i] = left_product[i-1] * nums[i-1];
        }
        for (int i =n-2;i>-1;i--)
        {
            right_product[i] = right_product[i+1] * nums[i+1];
        }
        vector<int> result(n,1);
        for (int i =0;i<n;i++)
        {
            result[i] = left_product[i] * right_product[i];
        }
        return result;
    }
};

File: .py/238.ProductofArrayExceptSelf.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        left_product = [1] * n
        right_product = [1]* n

        for i in range(1,n):
            left_product[i] = nums[i-1] * left_product[i-1]

        for i in range(n-2,-1,-1):
            right_product[i] = nums[i+1] * right_product[i+1]

        product = [left_product[i] * right_product[i] for i in range(n)]
        return product
        